crates/nri-resctrl-plugin/src/lib.rs

```diff
@@ -244,6 +466,61 @@ impl<P: FsProvider + Send + Sync + 'static> Plugin for ResctrlPlugin<P> {
         req: StateChangeEvent,
     ) -> ttrpc::Result<Empty> {
         debug!("resctrl-plugin: state_change: event={:?}", req.event);
+        match req.event.enum_value() {
+            Ok(Event::REMOVE_POD_SANDBOX) => {
+                if let Some(pod) = req.pod.as_ref() {
+                    let pod_uid = pod.uid.clone();
+                    let mut st = self.state.lock().unwrap();
+                    let group_path = st
+                        .pods
+                        .get(&pod_uid)
+                        .and_then(|ps| ps.group_path.clone());
+                    st.containers.retain(|_, c| c.pod_uid != pod_uid);
+                    st.pods.remove(&pod_uid);
+                    drop(st);
+                    if let Some(gp) = group_path {
+                        if let Err(e) = self.resctrl.delete_group(&gp) {
+                            warn!("resctrl-plugin: failed to delete group {}: {}", gp, e);
+                        }
+                        self.emit_event(PodResctrlEvent::Removed(PodResctrlRemoved {
+                            pod_uid,
+                            group_path: Some(gp),
```

Comment:
we don't need to repeat the group_path in the Removed message, let's simplify it